server/controllers/reservationController.js

```diff
@@ -168,55 +169,28 @@ const updateReservation = async (query) => {
 
 const sendEmail = async (reservationId) => {
   try {
-    const { data: combinedData, error: combinedError } = await supabase
-      .from("reservations")
+    const fullPath = await generateAndStoreQRCode(reservationId);
+
+    const [{ data: combinedData, error: combinedError }, { data: qrData, error: qrError }] = await Promise.all([
+      supabase.from("reservations")
       .select(
-        `
-        id,
+        `id,
         parking_spot_id,
         date,
         time,
         status,
         user_id:accounts(id, email),
-        garage_id:garages(id, name, address, city, state, country, zip)
-      `
+        garage_id:garages(id, name, address, city, state, country, zip)`
       )
       .eq("id", reservationId)
-      .single();
+      .single(),
+      supabase.storage.from("qrcodes").getPublicUrl(`${reservationId}.png`)
+    ])
 
-    if (combinedError) {
-      console.error("Error querying Supabase:", combinedError.message);
+    if (combinedError || qrError) {
       throw new Error("Internal Server Error");
     }
 
```

Comment:
thanks or integrating kyle's QR Generator controller into the the email transporter